### DISCUSSION TITLE
Consider dropping yaml support?

### DISCUSSION THREAD

**Discussion by jimmywarting** [eslint/eslint#15413] [Ideas] [answered]
I know this might be a big ask, but would you consider dropping support for yaml?
eslint have grown a lot in size and complexity 
I think yaml is just an extra overhead that isn't so beneficial. 

- Parsing it requires a extra module
- JSON is as simple as using `JSON.parse(str)` and require no dependencies
- And a JavaScript file that exports the object is also just as simple as importing it. 
  - it's much simpler to load as js file than a json atm if you look at it from a esm perspective where `import(json)` don't currently work and requires fs, readFile, and strip BOM to work.

having more option to solve a problem only increase the complexity of the application

I know a lot of folks might currently depend on yaml, but a deprecation would at least be a start...

**Comment by nzakas** [comment, marked as answer]
The new config system doesn’t support YAML files so we can remove YAML completely when we remove the current config system.

**Comment by planeiii-te** [comment]
Typing YAML is far superior to typing JSON was the biggest draw. JSON is the XML of the config world in terms of verbosity. Config files are edited by humans. Making their job easier seems like the biggest win. Sad to see it go.
